Sources/Commands/SwiftTestTool.swift

```diff
@@ -196,21 +196,36 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
     ///
     /// - Returns: The path to the test binary.
     private func buildTestsIfNeeded(_ options: TestToolOptions) throws -> AbsolutePath {
-        let buildPlan = try BuildPlan(buildParameters: self.buildParameters(), graph: loadPackageGraph())
-        if options.shouldBuildTests {
-            try build(plan: buildPlan, subset: .allIncludingTests)
+        // Check if we need to create the build plan.
+        if try shouldRegenerateManifest() || toolCache.testBinary == nil {
+            let buildPlan = try BuildPlan(buildParameters: buildParameters(), graph: loadPackageGraph())
+            if options.shouldBuildTests {
+                try self.build(plan: buildPlan, subset: .allIncludingTests)
+            }
+            guard let testProduct = buildPlan.buildProducts.first(where: { $0.product.type == .test }) else {
+                throw TestError.testsExecutableNotFound
+            }
+
+            // Go up the folder hierarchy until we find the .xctest bundle.
+            let testBinary = sequence(first: testProduct.binary, next: {
```

Comment:
Its a bundle on macOS and executable on Linux so we just call it binary.

Reply:
Ah okay, I'm sure it's already had a lot of discussion and changing that naming convention is out of scope of this PR.

---

Perhaps `ResolvedProduct` could have a computed property "`path`" that points to the app bundle, test bundle, or binary, depending on the product type. That would also address the sentiment of my query below:
> Would it make more sense to put the logic to update toolCache in SwiftTool so it can be called from build(plan:subset:)?